sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -220,6 +248,39 @@ def test_scorer(estimator, X, y):
     assert_array_equal(X_r_sparse.toarray(), iris.data)
 
 
+def test_rfecv_regression():
+    X, y, coef = make_regression(n_samples=100, n_features=10,
+                                 n_informative=2, coef=True, random_state=17)
+    X_true = X[:, coef > 0]
+    rfecv = RFECV(estimator=LinearRegression(), step=1, cv=5)
+    rfecv.fit(X, y)
+
+    assert_equal(len(rfecv.grid_scores_), X.shape[1])
+    assert_equal(len(rfecv.ranking_), X.shape[1])
+
+    X_r = rfecv.transform(X)
+    assert_array_almost_equal(X_r, X_true)
+
+    rfecv_sparse = RFECV(estimator=LinearRegression(), step=1, cv=5)
+    X_sparse = sparse.csr_matrix(X)
+    rfecv_sparse.fit(X_sparse, y)
+    X_r_sparse = rfecv_sparse.transform(X_sparse)
+    assert_array_almost_equal(X_r_sparse.toarray(), X_true)
+
+    rfecv = RFECV(estimator=LinearRegression(), step=4, cv=5)
+    rfecv.fit(X, y)
+    assert_equal(len(rfecv.grid_scores_), 4)
+    assert_equal(len(rfecv.ranking_), X.shape[1])
+    X_r = rfecv.transform(X)
+    assert_array_equal(X_r, X_true)
+
+    rfecv_sparse = RFECV(estimator=LinearRegression(), step=2, cv=5)
+    X_sparse = sparse.csr_matrix(X)
+    rfecv_sparse.fit(X_sparse, y)
+    X_r_sparse = rfecv_sparse.transform(X_sparse)
+    assert_array_equal(X_r_sparse.toarray(), X_true)
```

Comment:
you should factorize these two tests either with an aux function _test_rfe or with a for loop looping on a regressor and a classifier.